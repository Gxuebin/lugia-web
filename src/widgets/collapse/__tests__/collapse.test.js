/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import chai from 'chai';

import 'jest-styled-components';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CollapseDemo } from '../demo';
import Collapse from '../collapse';
import Panel from '../panel';
import renderer from 'react-test-renderer';
import { delay } from '@lugia/react-test-utils';

const { expect: exp } = chai;

Enzyme.configure({ adapter: new Adapter() });

describe('Collapse', () => {
  const createTarget = (props: Object): any => {
    return (
      <Collapse {...props}>
        <Panel value="1">
          <div>PanelContent...</div>
          <div>PanelContent...</div>
        </Panel>
        <Panel value="2">
          <div>PanelContent...</div>
          <div>PanelContent...</div>
        </Panel>
      </Collapse>
    );
  };
  const getState = (target: Object): Object => {
    return target
      .children()
      .at(0)
      .instance().state;
  };

  it('Collapse CSS', () => {
    const Target = <CollapseDemo />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });

  it('Collapse props:{activeValue} toTest state: value', () => {
    const props = {
      activeValue: '1',
    };
    const target = mount(createTarget(props));
    const state = getState(target);
    expect(state.value).toEqual(['1']);

    target.setProps({ activeValue: '2' });
    const newState = getState(target);
    expect(newState.value).toEqual(['2']);
  });

  it('Collapse props:{defaultActiveValue} toTest state: value', () => {
    const props = {
      defaultActiveValue: '1',
    };
    const target = mount(createTarget(props));
    const state = getState(target);
    expect(state.value).toEqual(['1']);

    target.setProps({ defaultActiveValue: '2' });
    const newState = getState(target);
    expect(newState.value).toEqual(['1']);
  });

  it('Collapse props:{defaultActiveValue} toTest onChange -> state: value', async () => {
    const props = {
      defaultActiveValue: '1',
    };
    const target = mount(createTarget(props));
    const state = getState(target);
    expect(state.value).toEqual(['1']);

    const panel = target.find('Panel');
    panel.at(1).simulate('click');
    await delay(1000);
    const newState = getState(target);
    expect(newState.value).toEqual(['1', '2']);
  });
  it('Collapse props:{defaultActiveValue} toTest onChange params(newValue,oldValue) ', async () => {
    let newValues, oldValues;
    const handleChange = ({ newValue, oldValue }: Object) => {
      newValues = newValue;
      oldValues = oldValue;
    };
    const props = {
      defaultActiveValue: '1',
      onChange: handleChange,
    };
    const target = mount(createTarget(props));
    const panel = target.find('Panel');

    panel.at(1).simulate('click');
    await delay(1000);
    expect(newValues).toEqual(['1', '2']);
    expect(oldValues).toEqual(['1']);

    panel.at(0).simulate('click');
    await delay(1000);
    expect(newValues).toEqual(['2']);
    expect(oldValues).toEqual(['1', '2']);
  });

  it('Collapse props:{accordion} toTest onChange state(value) ', async () => {
    const props = {
      defaultActiveValue: '1',
      accordion: true,
    };
    const target = mount(createTarget(props));
    const state = getState(target);
    expect(state.value).toEqual(['1']);

    const panel = target.find('Panel');

    panel.at(0).simulate('click');
    await delay(1000);
    const states = getState(target);
    expect(states.value).toEqual([]);

    panel.at(1).simulate('click');
    await delay(1000);
    const newState = getState(target);
    expect(newState.value).toEqual(['2']);

    panel.at(1).simulate('click');
    await delay(1000);
    const newState2 = getState(target);
    expect(newState2.value).toEqual([]);
  });

  it('Collapse props:{accordion} toTest onChange params(newValue,oldValue) ', async () => {
    let newValues, oldValues;
    const handleChange = ({ newValue, oldValue }: Object) => {
      newValues = newValue;
      oldValues = oldValue;
    };
    const props = {
      defaultActiveValue: '1',
      accordion: true,
      onChange: handleChange,
    };
    const target = mount(createTarget(props));
    const state = getState(target);
    expect(state.value).toEqual(['1']);

    const panel = target.find('Panel');

    panel.at(0).simulate('click');
    await delay(1000);
    expect(newValues).toEqual([]);
    expect(oldValues).toEqual(['1']);

    panel.at(0).simulate('click');
    await delay(1000);
    expect(newValues).toEqual(['1']);
    expect(oldValues).toEqual([]);

    panel.at(1).simulate('click');
    await delay(1000);
    expect(newValues).toEqual(['2']);
    expect(oldValues).toEqual(['1']);
  });
});
