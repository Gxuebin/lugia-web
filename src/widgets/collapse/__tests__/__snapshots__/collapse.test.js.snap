// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Collapse CSS 1`] = `
.c3 {
  box-sizing: border-box;
  background: #fff;
  border-color: #e8e8e8;
  border-style: solid;
  border-width: 0 0 1px;
}

.c4 {
  box-sizing: border-box;
  position: relative;
  font-size: 1.4rem;
  cursor: pointer;
  line-height: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  color: #333333;
  opacity: 1;
}

.c6 {
  color: #333333;
  font-size: 1.4rem;
}

.c7 {
  box-sizing: border-box;
  overflow: hidden;
  height: 100%;
}

.c10 {
  box-sizing: border-box;
  overflow: hidden;
  height: 0rem;
}

.c8 {
  box-sizing: border-box;
  line-height: 1.5;
  color: #666666;
  font-size: 1.4rem;
  font-weight: 300;
  background-color: #fff;
}

.c2 {
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  font-size: 1.4rem;
}

.c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  position: absolute;
  top: 50%;
  left: 1rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-transform: translateY(-50%) rotate(90deg);
  -ms-transform: translateY(-50%) rotate(90deg);
  transform: translateY(-50%) rotate(90deg);
}

.c9 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  position: absolute;
  top: 50%;
  left: 1rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c1 {
  font-size: 1.4rem;
  opacity: 1;
}

.c0 {
  width: 900px;
  margin: 0 auto;
  padding: 50px;
  border: 1px solid #e8e8e8;
}

<div>
  <div
    className="c0"
  >
    <p>
      defaultActiveValue=
      1
    </p>
    <div
      className="c1"
      style={
        Object {
          "backgroundColor": "orange",
          "borderBottomColor": "green",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0.2rem",
          "borderLeftColor": "green",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0.2rem",
          "borderRightColor": "green",
          "borderRightStyle": "solid",
          "borderRightWidth": "0.2rem",
          "borderTopColor": "green",
          "borderTopStyle": "solid",
          "borderTopWidth": "0.2rem",
          "height": "50rem",
          "marginBottom": "1rem",
          "marginLeft": "1rem",
          "marginRight": "1rem",
          "marginTop": "1rem",
          "opacity": 1,
          "paddingBottom": "1rem",
          "paddingLeft": "1rem",
          "paddingRight": "1rem",
          "paddingTop": "1rem",
          "width": "41rem",
        }
      }
    >
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "pink",
                "borderBottomColor": "yellow",
                "borderBottomLeftRadius": "1rem",
                "borderBottomRightRadius": "1rem",
                "borderBottomStyle": "solid",
                "borderBottomWidth": "0.2rem",
                "borderLeftColor": "yellow",
                "borderLeftStyle": "solid",
                "borderLeftWidth": "0.2rem",
                "borderRightColor": "yellow",
                "borderRightStyle": "solid",
                "borderRightWidth": "0.2rem",
                "borderTopColor": "yellow",
                "borderTopLeftRadius": "1rem",
                "borderTopRightRadius": "1rem",
                "borderTopStyle": "solid",
                "borderTopWidth": "0.2rem",
                "boxShadow": "1px 2px 2px 2px #e8e8e8",
                "height": "8rem",
                "opacity": 1,
                "paddingBottom": "1rem",
                "paddingLeft": "3rem",
                "paddingRight": "1rem",
                "paddingTop": "1rem",
                "width": "40rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c5"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "green",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={
                Object {
                  "color": "green",
                  "fontSize": "1.6rem",
                  "fontWeight": 300,
                }
              }
            >
              LUGIA
            </span>
          </div>
          <div
            className="c7"
            height={0}
            open={true}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "backgroundColor": "#f7f4ef",
                  "color": "red",
                  "fontSize": "1.6rem",
                  "fontWeight": 300,
                  "height": "20rem",
                  "paddingBottom": "1rem",
                  "paddingLeft": "3rem",
                  "paddingRight": "1rem",
                  "paddingTop": "1rem",
                  "width": "40rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "pink",
                "borderBottomColor": "yellow",
                "borderBottomLeftRadius": "1rem",
                "borderBottomRightRadius": "1rem",
                "borderBottomStyle": "solid",
                "borderBottomWidth": "0.2rem",
                "borderLeftColor": "yellow",
                "borderLeftStyle": "solid",
                "borderLeftWidth": "0.2rem",
                "borderRightColor": "yellow",
                "borderRightStyle": "solid",
                "borderRightWidth": "0.2rem",
                "borderTopColor": "yellow",
                "borderTopLeftRadius": "1rem",
                "borderTopRightRadius": "1rem",
                "borderTopStyle": "solid",
                "borderTopWidth": "0.2rem",
                "boxShadow": "1px 2px 2px 2px #e8e8e8",
                "height": "8rem",
                "opacity": 1,
                "paddingBottom": "1rem",
                "paddingLeft": "3rem",
                "paddingRight": "1rem",
                "paddingTop": "1rem",
                "width": "40rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c9"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "green",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={
                Object {
                  "color": "green",
                  "fontSize": "1.6rem",
                  "fontWeight": 300,
                }
              }
            >
              LUGIA
            </span>
          </div>
          <div
            className="c10"
            height={0}
            open={false}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "backgroundColor": "#f7f4ef",
                  "color": "red",
                  "fontSize": "1.6rem",
                  "fontWeight": 300,
                  "height": "20rem",
                  "paddingBottom": "1rem",
                  "paddingLeft": "3rem",
                  "paddingRight": "1rem",
                  "paddingTop": "1rem",
                  "width": "40rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br />
    <p>
      defaultActiveValue=
      1
    </p>
    <div
      className="c1"
      style={Object {}}
    >
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c5"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c7"
            height={0}
            open={true}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c9"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c10"
            height={0}
            open={false}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br />
    <p>
      activeValue=
      1
    </p>
    <div
      className="c1"
      style={Object {}}
    >
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c5"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c7"
            height={0}
            open={true}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c9"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c10"
            height={0}
            open={false}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br />
    <p>
      activeValue=
      this.state.activeValue
    </p>
    <div
      className="c1"
      style={Object {}}
    >
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c5"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c7"
            height={0}
            open={true}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c9"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c10"
            height={0}
            open={false}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br />
    <p>
      accordion
    </p>
    <div
      className="c1"
      style={Object {}}
    >
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c5"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c7"
            height={0}
            open={true}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c9"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c10"
            height={0}
            open={false}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c9"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c10"
            height={0}
            open={false}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c9"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c10"
            height={0}
            open={false}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br />
    <p>
      data
    </p>
    <div
      className="c1"
      style={Object {}}
    >
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c5"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c7"
            height={0}
            open={true}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              PanelContent...
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c9"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c10"
            height={0}
            open={false}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              PanelContent...
            </div>
          </div>
        </div>
      </div>
    </div>
    <p>
      collapse nest
    </p>
    <div
      className="c1"
      style={Object {}}
    >
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c5"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c7"
            height={0}
            open={true}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div
                className="c1"
                style={Object {}}
              >
                <div
                  className="c2"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={Object {}}
                >
                  <div
                    className="c3"
                  >
                    <div
                      className="c4"
                      disabled={false}
                      onClick={[Function]}
                      style={
                        Object {
                          "paddingBottom": "1.6rem",
                          "paddingLeft": "3rem",
                          "paddingRight": "0rem",
                          "paddingTop": "1.6rem",
                        }
                      }
                    >
                      <i
                        className="lugia-icon-direction_caret_right   c5"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        style={
                          Object {
                            "color": "#666",
                            "fontSize": "1.4rem",
                          }
                        }
                      />
                      <span
                        className="c6"
                        style={Object {}}
                      >
                        LUGIA
                      </span>
                    </div>
                    <div
                      className="c7"
                      height={0}
                      open={true}
                      style={Object {}}
                    >
                      <div
                        className="c8"
                        style={
                          Object {
                            "paddingBottom": "2.2rem",
                            "paddingLeft": "3.4rem",
                            "paddingRight": "3rem",
                            "paddingTop": "0.6rem",
                          }
                        }
                      >
                        <div>
                          PanelContent...
                        </div>
                        <div>
                          PanelContent...
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="c2"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={Object {}}
                >
                  <div
                    className="c3"
                  >
                    <div
                      className="c4"
                      disabled={false}
                      onClick={[Function]}
                      style={
                        Object {
                          "paddingBottom": "1.6rem",
                          "paddingLeft": "3rem",
                          "paddingRight": "0rem",
                          "paddingTop": "1.6rem",
                        }
                      }
                    >
                      <i
                        className="lugia-icon-direction_caret_right   c9"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        style={
                          Object {
                            "color": "#666",
                            "fontSize": "1.4rem",
                          }
                        }
                      />
                      <span
                        className="c6"
                        style={Object {}}
                      >
                        LUGIA
                      </span>
                    </div>
                    <div
                      className="c10"
                      height={0}
                      open={false}
                      style={Object {}}
                    >
                      <div
                        className="c8"
                        style={
                          Object {
                            "paddingBottom": "2.2rem",
                            "paddingLeft": "3.4rem",
                            "paddingRight": "3rem",
                            "paddingTop": "0.6rem",
                          }
                        }
                      >
                        <div>
                          PanelContent...
                        </div>
                        <div>
                          PanelContent...
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c2"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="c3"
        >
          <div
            className="c4"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "paddingBottom": "1.6rem",
                "paddingLeft": "3rem",
                "paddingRight": "0rem",
                "paddingTop": "1.6rem",
              }
            }
          >
            <i
              className="lugia-icon-direction_caret_right   c9"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "#666",
                  "fontSize": "1.4rem",
                }
              }
            />
            <span
              className="c6"
              style={Object {}}
            >
              LUGIA
            </span>
          </div>
          <div
            className="c10"
            height={0}
            open={false}
            style={Object {}}
          >
            <div
              className="c8"
              style={
                Object {
                  "paddingBottom": "2.2rem",
                  "paddingLeft": "3.4rem",
                  "paddingRight": "3rem",
                  "paddingTop": "0.6rem",
                }
              }
            >
              <div>
                PanelContent...
              </div>
              <div>
                PanelContent...
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
